### DISCUSSION TITLE
anomaly image saving

### DISCUSSION THREAD

**Discussion by mouxinyue1** [open-edge-platform/anomalib#899] [Q&A]
Hi, I intend to save abnormal image separately without superimposing the original image, which is used as lightning_inference.py. What kind of modification do I need

**Comment by samet-akcay** [comment]
@mouxinyue1, you could use `torch_inference.py`, which would give you more freedom. 
https://github.com/openvinotoolkit/anomalib/blob/879e2f81849095c36104ede077b73a347b865928/tools/inference/torch_inference.py#L91

`predictions` object contains all the information you need such as image, true_label, true_mask, pred_label, pred_mask, anomaly_map, heat_map. You could modify it however you want.

**Reply by mouxinyue1** [reply]
I tried torch_inference.py. But the following error will be reported. I don't know how to solve it
![image](https://user-images.githubusercontent.com/91049826/217697639-c8ce7cd8-1c69-406a-9794-89f4e8deb30d.png)
When I remove this parameter, I get a new error
![image](https://user-images.githubusercontent.com/91049826/217699638-09e4fcc5-1977-4b0d-8a5b-ef576e477555.png)
